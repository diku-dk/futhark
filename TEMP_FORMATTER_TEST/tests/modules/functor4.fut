-- Another parametric module where the argument contains an abstract type.
-- ==
-- input {} output {2}

module type foo = {
  type foo

  val mkfoo: i32 -> foo
}

module rgba_foo: foo = {
  type foo = i32

  def mkfoo (x: i32) = x
}

module foospace(C: foo) = {
  open C

  def frob (x: foo): foo = x
}

module rgba = foospace(rgba_foo)

def main = 2
