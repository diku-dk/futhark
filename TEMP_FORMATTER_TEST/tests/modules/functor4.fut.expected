da99f4a8a247ff1e681e5f92e0fb35af
